src/endpoint/s3/s3_bucket_policy_utils.js

```diff
@@ -149,20 +149,21 @@ async function _is_object_version_fit(req, predicate, value) {
     return res;
 }
 
-async function has_bucket_policy_permission(policy, account, method, arn_path, req, disallow_public_access = false) {
+async function has_bucket_policy_permission(policy, account, method, arn_path, req, disallow_public_access = false,
+        should_pass_principal = true) {
```

Comment:
Giving 2 options with Boolean default value can lead to bugs
```suggestion
async function has_bucket_policy_permission(policy, account, method, arn_path, req, 
{
disallow_public_access = false,
    should_pass_principal = true,
  } = {}
```
Then, when calling it, you can: 
```
has_bucket_policy_permission(policy, account, method, arn, req, {
  should_pass_principal: true,
});
```